cou/apps/auxiliary.py

```diff
@@ -183,6 +183,87 @@ class RabbitMQServer(AuxiliaryApplication):
     # COU changes to 3.9 if the channel is set to 3.8
     multiple_channels = True
 
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PreUpgradeStep]:
+        """Pre Upgrade steps planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of pre upgrade steps.
+        :rtype: list[PreUpgradeStep]
+        """
+        steps = super().pre_upgrade_steps(target, units)
+        if self.config["enable-auto-restarts"].get("value") is False:
```

Comment:
Done

Reply:
Oh no I realize that you are doing `self.config["enable-auto-restarts"]`, I think it would be safer to use get, because there is maybe version without it. This is done only for rabbitmq class, but still this can cause unexcpected error.

Also checking if value is False in that case is needed, since there could be None return.
```if self.config.get("enable-auto-restarts", {}).get("value") is False:```

What do you say?

Reply:
this pattern in multiple place, should we open seperate PR for that alone?

Reply:
Issue opened: https://github.com/canonical/charmed-openstack-upgrader/issues/426

Reply:
closing for now

Reply:
- cou/apps/base.py:113 no proper example, since we are iterationg over config
- cou/apps/base.py:239 no proper example, since origin_setting check if that exists
- cou/apps/base.py:696 no proper example, since there is check in line 690
- cou/apps/base.py:858 no proper example, since there is check in line 852
- cou/apps/auxiliary.py:264 no proper example, since there is check in line 258


The issue is that you change can cause error KeyError and that would not give much information to user + wrong user experience.